src/Glpi/Form/QuestionType/QuestionTypeUserDevice.php

```diff
@@ -0,0 +1,268 @@
+<?php
+
+/**
+ * ---------------------------------------------------------------------
+ *
+ * GLPI - Gestionnaire Libre de Parc Informatique
+ *
+ * http://glpi-project.org
+ *
+ * @copyright 2015-2024 Teclib' and contributors.
+ * @copyright 2003-2014 by the INDEPNET Development Team.
+ * @licence   https://www.gnu.org/licenses/gpl-3.0.html
+ *
+ * ---------------------------------------------------------------------
+ *
+ * LICENSE
+ *
+ * This file is part of GLPI.
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <https://www.gnu.org/licenses/>.
+ *
+ * ---------------------------------------------------------------------
+ */
+
+namespace Glpi\Form\QuestionType;
+
+use CommonItilObject_Item;
+use Glpi\Application\View\TemplateRenderer;
+use Glpi\Form\Question;
+use Override;
+use Session;
+
+final class QuestionTypeUserDevice extends AbstractQuestionType
+{
+    #[Override]
+    public function validateExtraDataInput(array $input): bool
+    {
+        $allowed_keys = [
+            'is_multiple_devices'
+        ];
+
+        return empty(array_diff(array_keys($input), $allowed_keys))
+            && array_reduce($input, fn($carry, $value) => $carry && preg_match('/^[01]$/', $value), true);
```

Comment:
I had to read twice to understand. It seems pretty complex to just validate that only the `is_multiple_devices` is allowed and that the value must be a boolean.